### DISCUSSION TITLE
PHP Deprecated

### DISCUSSION THREAD

**Discussion by octoberweb69** [craigk5n/webcalendar#460] [General]
After much effort, I managed to install WebCalendar 1.9.10 on a Windows server with IIS 10, PHP 8.3.2, MySQL 8.0.36. When I input an event and view the calendar, the following errors appear:

`PHP Deprecated:  Creation of dynamic property Event::$_due is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 259
PHP Deprecated:  Creation of dynamic property Event::$_DateTime is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 426
PHP Deprecated:  Creation of dynamic property Event::$_due is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 259
PHP Deprecated:  Creation of dynamic property Event::$_DateTime is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 426
PHP Deprecated:  Creation of dynamic property Event::$_due is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 259
PHP Deprecated:  Creation of dynamic property Event::$_DateTime is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 426
PHP Deprecated:  Creation of dynamic property Event::$_Date is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 316
PHP Deprecated:  Creation of dynamic property Event::$_Date is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 316
PHP Deprecated:  Creation of dynamic property Event::$_Date is deprecated in C:\DATA\WWW\WebCalendar\includes\classes\Event.php on line 316`

How can I solve the issue?
Thanks

**Comment by bryanvt** [comment]
I don't know if you found a solution to this problem. But in case not...

I was experiencing the same thing. (Odd because it had been working fine prior to a reboot.) Anyway, after spending some time trying to track it down, I decided to just bring the code up to date. Most of the properties, the exception being _due in Events.php, were defined within methods and returned. The properties as defined on $this were only used within the methods. So it was trivial to modify the methods to use a local var, and return, rather than a dynamic property on the class itself.

Hope that helps.
